rust/signed_doc/src/lib.rs

```diff
@@ -105,6 +106,177 @@ impl CatalystSignedDocument {
     pub fn signatures(&self) -> &Signatures {
         &self.inner.signatures
     }
+
+    /// Verify document signatures and `UUID`s.
+    ///
+    /// # Errors
+    ///
+    /// Returns a report of verification failures and the source error.
+    #[allow(clippy::indexing_slicing, clippy::too_many_lines)]
+    pub fn verify<P>(&self, pk_getter: P) -> Result<(), CatalystSignedDocError>
+    where P: Fn(&KidUri) -> VerifyingKey {
+        let error_report = ProblemReport::new("Catalyst Signed Document Verification");
+
+        if !self.doc_type().is_valid() {
```

Comment:
We've updated a `UuidV4` and `UuidV7` decoding logic, so now we can assume that they always valid.
So such validation is not necessary at this place, because it is already done during decoding